src/components/TitleTransferPanel/Actions/FinaliseEndorsement.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { AddressesInputs } from "./../AddressInput";
+
+interface FinaliseEndorsementProps {
+  finaliseEndorsementHolderAddress: string;
+  finaliseEndorsementBeneficiaryAddress: string;
+}
+
+export const FinaliseEndorsement = ({
+  finaliseEndorsementHolderAddress,
+  finaliseEndorsementBeneficiaryAddress
+}: FinaliseEndorsementProps) => {
+  return (
+    <AddressesInputs
```

Comment:
hmmm why combine the two fields into one component? would it work better if we just have two \<AddressInput\> here?

Reply:
just thinking maybe we can have a TitleEscrow component
```
<TitleEscrow contractAddress="xx" />
```

And let whatever is inside TitleEscrow figure out how to retrieve the addresses and fire the actions - then we can reuse this \<TitleEscrow\> component in PrepareEndorsement (editable), RemoteEndorsement (editable) and FinaliseEndorsement (view only)